### ISSUE TITLE
Add notes about "how to fix vulnerable dependencies"

### ISSUE THREAD

**Issue by bhamail** [sonatype-nexus-community/nancy#152]
During @DarthHater 's webcast, he walked through the process of how to fix a vulnerablility detected by Nancy.

We should add some short notes about those steps and the nifty related go mod commands, e.g
How to view which dependencies are pulling in a specific library.
How to research the available versions of a library.
How to see the "newest" available version of a library. etc.

cc @bhamail / @DarthHater

**Comment by bhamail** [comment]
Example:

1. Nancy finds a vulnerability. In this example, a vulnerability was found in the `etcd` package:

```
$ go list -json -m all | nancy sleuth
...
[CVE-2020-15114] In etcd before versions 3.3.23 and 3.4.10, the etcd gateway is a simple TCP prox...
```

2. The vulnerability is not in a "direct" dependency (which would be listed in the `go.mod` file), but rather the vulnerability is in a transitive dependency (a dependency of a dependency). You can use the `go mod graph` command to print  transitive dependencies. For example, the command below will show which module pulls in the vulnerable `etcd` package detected by nancy:
```
$ go mod graph | grep coreos/etcd
github.com/bketelsen/crypt@v0.0.3-0.20200106085610-5cbc8cc4026c github.com/coreos/etcd@v3.3.13+incompatible
```

3. You can use a `replace` directive in the `go.mod` file to use a newer version of the transitive dependency. See the "replace" directive section of [Go Modules Reference](https://golang.org/ref/mod#go). To avoid semver issues, you probably want to use a newer dependency version that is in the same "major.minor" version as the vulnerable dependency version.
Here is an example `replace` added to a `go.mod` file.
```
// fix vulnerability: CVE-2020-15114 in etcd v3.3.13+incompatible
replace github.com/coreos/etcd => github.com/coreos/etcd v3.3.24+incompatible
```

Be aware that even after you add a `replace` directive, the `go mod graph` will show the old dependency version. You can verify the new version is actually used via the `go list` command:
```
$ go list -m all | grep coreos/etcd
github.com/coreos/etcd v3.3.13+incompatible => github.com/coreos/etcd v3.3.24+incompatible
```
You can see the v3.3.13 is replaced with v3.3.24.

4. Submit a PR to the project with the vulnerable dependency to let them know about the issue and ideally fix it in a new release of that dependency. Even better, also tell them about `nancy` and maybe then will add `nancy` to their own CI system.

**Comment by bhamail** [comment]
@churchryan @guillermo-varela and I discussed adding a simple message to all the tooling, for example:
When a vulnerability exists, also print a message showing how to print a dependency tree for that environment. 
e.g. for gradle: `gradle dependencies`
